tests/test_peer_did_resolving.py

```diff
@@ -0,0 +1,69 @@
+                   "type": "ED25519",
+                   "controller": "did:peer:2.Ez6LSbysY2xFMRpGMhb7tFTLMpeuPRaqaWM1yECx2AtzE3KCc.Vz6MkqRYqQiSgvZQdnBytw86Qbs2ZWUkGv22od935YF4s8M7V.Vz6MkgoLTnTypo3tDRwCkZXSccTPHRLhF4ZnjhueYAFpEX6vg.SeyJ0IjoiZG0iLCJzIjoiaHR0cHM6Ly9leGFtcGxlLmNvbS9lbmRwb2ludCIsInIiOlsiZGlkOmV4YW1wbGU6c29tZW1lZGlhdG9yI3NvbWVrZXkiXX0=",
+                   "publicKeyBase58": "3M5RCDjPTWPkKSN3sxUmmMqHbmRPegYP1tjcKyrDbt9J"
+               }
+           ],
+           "keyAgreement": [
+               {
+                   "id": "did:peer:2.Ez6LSbysY2xFMRpGMhb7tFTLMpeuPRaqaWM1yECx2AtzE3KCc.Vz6MkqRYqQiSgvZQdnBytw86Qbs2ZWUkGv22od935YF4s8M7V.Vz6MkgoLTnTypo3tDRwCkZXSccTPHRLhF4ZnjhueYAFpEX6vg.SeyJ0IjoiZG0iLCJzIjoiaHR0cHM6Ly9leGFtcGxlLmNvbS9lbmRwb2ludCIsInIiOlsiZGlkOmV4YW1wbGU6c29tZW1lZGlhdG9yI3NvbWVrZXkiXX0=#6LSbysY2xFMRpGMhb7tFTLMpeuPRaqaWM1yECx2AtzE3KCc",
+                   "type": "X25519",
+                   "controller": "did:peer:2.Ez6LSbysY2xFMRpGMhb7tFTLMpeuPRaqaWM1yECx2AtzE3KCc.Vz6MkqRYqQiSgvZQdnBytw86Qbs2ZWUkGv22od935YF4s8M7V.Vz6MkgoLTnTypo3tDRwCkZXSccTPHRLhF4ZnjhueYAFpEX6vg.SeyJ0IjoiZG0iLCJzIjoiaHR0cHM6Ly9leGFtcGxlLmNvbS9lbmRwb2ludCIsInIiOlsiZGlkOmV4YW1wbGU6c29tZW1lZGlhdG9yI3NvbWVrZXkiXX0=",
+                   "publicKeyBase58": "JhNWeSVLMYccCk7iopQW4guaSJTojqpMEELgSLhKwRr"
+               }
+           ],
+           "service": [
+               {
+                   "id": "did:peer:2.Ez6LSbysY2xFMRpGMhb7tFTLMpeuPRaqaWM1yECx2AtzE3KCc.Vz6MkqRYqQiSgvZQdnBytw86Qbs2ZWUkGv22od935YF4s8M7V.Vz6MkgoLTnTypo3tDRwCkZXSccTPHRLhF4ZnjhueYAFpEX6vg.SeyJ0IjoiZG0iLCJzIjoiaHR0cHM6Ly9leGFtcGxlLmNvbS9lbmRwb2ludCIsInIiOlsiZGlkOmV4YW1wbGU6c29tZW1lZGlhdG9yI3NvbWVrZXkiXX0=#didcommmessaging",
+                   "type": "didcommmessaging",
+                   "serviceEndpoint": "https://example.com/endpoint",
+                   "routingKeys": [
+                       "did:example:somemediator#somekey"
+                   ]
+               }
+           ]
+       }''')
+
+
+def test_resolve_numalgo_0_positive():
+    assert json.loads(resolve_peer_did('did:peer:0z6MkqRYqQiSgvZQdnBytw86Qbs2ZWUkGv22od935YF4s8M7V')) \
+           == \
+           json.loads('''{
+    "id": "did:peer:0z6MkqRYqQiSgvZQdnBytw86Qbs2ZWUkGv22od935YF4s8M7V",
+    "authentication": {
+        "id": "did:peer:0z6MkqRYqQiSgvZQdnBytw86Qbs2ZWUkGv22od935YF4s8M7V#6MkqRYqQiSgvZQdnBytw86Qbs2ZWUkGv22od935YF4s8M7V",
+        "type": "ED25519",
+        "controller": "did:peer:0z6MkqRYqQiSgvZQdnBytw86Qbs2ZWUkGv22od935YF4s8M7V",
+        "publicKeyBase58": "ByHnpUCFb1vAfh9CFZ8ZkmUZguURW8nSw889hy6rD8L7"
+    }
+}''')
+
+
+def test_resolve_wrong_peer_did():
```

Comment:
Please add more negative tests for resolving:
1) `test_resolve_unsupported_did_method` (a valid did, but not of peer did type)
2) `test_resolve_invalid_peer_did` (a invalid did of peer did type)
3) `test_resolve_unsupported_numalgo_code` (numalgo=1,3)
4) `test_resolve_numalgo_0_malformed_base58_encoding`
5) `test_resolve_numalgo_0_unsupported_transform_code` (not `z`)
6) `test_resolve_numalgo_0_malformed_multicodec_encoding` (not a valid multicodec encoding)
7) `test_resolve_numalgo_0_unsupported_key_type` (not from our Key enums)
8) `test_resolve_numalgo_0_invalid_key_type` (encryption instead of signing)
9) `test_resolve_numalgo_1_unsupported_transform_code` (not `z`)
10) `test_resolve_numalgo_1_signing_malformed_base58_encoding`
11) `test_resolve_numalgo_1_encryption_malformed_base58_encoding`
12) `test_resolve_numalgo_1_signing_malformed_multicodec_encoding` (not a valid multicodec encoding)
13) `test_resolve_numalgo_1_encryption_malformed_multicodec_encoding` (not a valid multicodec encoding)
14) `test_resolve_numalgo_1_signing_unsupported_key_type` (not from our Key enums)
15) `test_resolve_numalgo_1_encryption_unsupported_key_type` (not from our Key enums)
16) `test_resolve_numalgo_1_signing_invalid_key_type` (encryption instead of signing)
17) `test_resolve_numalgo_1_encryption_invalid_key_type` (signing instead of encryption)
18) `test_resolve_numalgo_1_malformed_service_encoding` (not base64)